src/plots/cartesian/axis_defaults.js

```diff
@@ -124,8 +125,9 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         }
     }
 
-    if(containerOut.type === 'date') {
-        var rangebreaks = containerIn.rangebreaks;
+    if(axType === 'date') {
+        var rangebreaks = containerIn.rangebreaks || axTemplate.rangebreaks;
```

Comment:
This isn't quite right... `[]` is truthy, also `axTemplate.rangeBreaks` items will extend the array if it contains named items.

Can we instead do:

```js
handleArrayContainerDefaults(...);

if (containerOut.rangebreaks.length) {
   // setConvert and gl / splom check
}
else {
   delete containerOut.rangebreaks;
}
```

Same for `tickformatstops`

Reply:
Good idea. Done in https://github.com/plotly/plotly.js/pull/4670/commits/f824902d0c044d3fcd7342b45efd20461248f6d1.